app/adapters/application.js

```diff
@@ -2,16 +2,29 @@ import Ember from 'ember';
 import DS from 'ember-data';
 import ENV from '../config/environment';
 
+// matches query params or trailing slash
+const TRAILING_SLASH_PATTERN = /(^.*?\?.*?|.*?[^\/]$)/;
+
 export default DS.RESTAdapter.extend(DS.BuildURLMixin, {
+    session: Ember.inject.service(),
     namespace: 'api/v2',
     host: ENV.apiBaseUrl,
     headers: {
         Accept: 'application/json'
     },
     ajax(url, method, hash) {
+        // trailing slash required for POST requests
+        if (method === 'POST') {
+            let urlRegex = new RegExp(TRAILING_SLASH_PATTERN);
+            if (url.match(urlRegex)) {
+                url = url + '/';
```

Comment:
Use [`String.replace`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace) with a regex here. Given a url with a query parameter the slash needs to be added before the `?`. If you do that you can ignore the method, the API want's trailing `/`s anyways.